spec/models/product_spec.rb

```diff
@@ -0,0 +1,142 @@
+require "rails_helper"
+
+RSpec.describe Product, type: :model do
+  subject {product}
+
+  let!(:supplier) { FactoryBot.create :supplier }
+  let!(:category) { FactoryBot.create :category }
+  let!(:product) {
+    FactoryBot.build :product,
+      supplier_id: supplier.id, category_id: category.id
+  }
+
+  describe "associations" do
+    it "belong to category" do
+      is_expected.to belong_to :category
+    end
+
+    it "belong to supplier" do
+      is_expected.to belong_to :supplier
+    end
+
+    it "has many image" do
+      is_expected.to have_many :images
+    end
+
+    it "has many order_details" do
+      is_expected.to have_many :order_details
+    end
+
+    it "has many comments" do
+      is_expected.to have_many :comments
+    end
+  end
+
+  describe "Delegate" do
+    it { is_expected.to delegate_method(:name).to(:category).with_prefix(true) }
+  end
+
+  describe "Validations" do
```

Comment:
describe, context, it để làm gì e /

Reply:
describe  là group các examle (trong các khối if). 
context để phân các example theo ngữ cảnh muốn test.
it là các example . các ví dụ test